controllers/api/comment-routes.js

```diff
@@ -1,12 +1,52 @@
 const router = require('express').Router();
-const { Comment } = require('../../models/');
+const { Comment, User, Post } = require('../../models/');
+const withAuth = require('../../utils/auth');
 
-router.get('/', async (req, res) => {
+//find all comments associated with a post = req.params.id
+router.get('/:id', withAuth, async (req, res) => {
   try {
-    const commentData = await Comment.findAll();
-    res.status(200).json(commentData);
+    //find post by query param
+    const commentData = await Post.findOne({
+      where: { id: req.params.id },
+      include: [
+        //include and user data for the post
+        {
+          model: User,
+          attributes: { exclude: ['password'] },
+        },
+        //include all comments for the post and the user data for those comments
+        {
+          model: Comment,
+          include: { model: User, attributes: { exclude: ['password'] } },
+        },
+      ],
+    });
+    const comment = commentData.get({ plain: true });
+    // render comment
+    res.json(comment);
+    //uncomment this when ready for front-end rendering
+    // res.render('comment', {
+    //   comment,
+    //   logged_in: req.session.logged_in,
+    // });
```

Comment:
If we're handling this commented out code in htmlRoutes, lets delete it here in apiRotes